### PR TITLE
Add ability to only push a subset of data during push

### PR DESCRIPTION
**Pull request by armandmcqueen**
This was tested manually. Integration tests should be added as well, but they will probably come via a different PR.

### CODE REVIEW

File: api/python/quilt3/packages.py [hunk 1]
```diff
@@ -1193,14 +1218,20 @@ def _materialize(self, dest_url):
         file_list = []
         entries = []
         for logical_key, entry in self.walk():
+
+            if selector_fn(logical_key, entry) == False:
+                # print(f"Skipping {logical_key}")
```
Comment (akarve) [inline api/python/quilt3/packages.py]:
```suggestion
```

File: api/python/quilt3/packages.py [hunk 2]
```diff
@@ -1193,14 +1218,20 @@ def _materialize(self, dest_url):
         file_list = []
         entries = []
         for logical_key, entry in self.walk():
+
+            if selector_fn(logical_key, entry) == False:
```
Comment (akarve) [inline api/python/quilt3/packages.py]: I don't understand why we set if the selector fails? we don't want `if selector_fn(...):`
```suggestion
            if not selector_fn(logical_key, entry):
```

File: api/python/quilt3/packages.py [hunk 3]
```diff
@@ -1193,14 +1218,20 @@ def _materialize(self, dest_url):
         file_list = []
         entries = []
         for logical_key, entry in self.walk():
+
```
Comment (akarve) [inline api/python/quilt3/packages.py]:
```suggestion
```

File: api/python/quilt3/packages.py (hunk 2 above)
Comment (armandmcqueen) [reply api/python/quilt3/packages.py]: At this point the new package is empty (see line 1215). We need to go through the entries in the old package and add them to the new package. If the selector_fn is false, we add it as is. If the selector_fn is true and the data isn't already in s3, we need to copy it to s3 and then add it to the new package (lines 1234-1235)

Comment (akarve) [reply api/python/quilt3/packages.py]: updated suggestion

**Pull request merged**